processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -863,8 +863,7 @@ public boolean hasEmptyAccessibleContructor() {
             hasEmptyAccessibleContructor = false;
             List<ExecutableElement> constructors = ElementFilter.constructorsIn( typeElement.getEnclosedElements() );
             for ( ExecutableElement constructor : constructors ) {
-                if ( (constructor.getModifiers().contains( Modifier.PUBLIC )
-                    || constructor.getModifiers().contains( Modifier.PROTECTED ) )
+                if ( !constructor.getModifiers().contains( Modifier.PRIVATE )
```

Comment:
How do we know here that the constructor is actually accessible from the package of the mapper?

Reply:
That is the case, we don't really know. And I think it is difficult to check this because the test is later on. Actually one of our test have package protected classes and package protected static classes. The modifiers are then empty (and that is why it started failing).

I am not sure if it is worth it to add more complex checks? It will create a compile problem in the generated file in any case.

Reply:
Sure, that’s okay as well.